pycqed/instrument_drivers/meta_instrument/device.py

```diff
@@ -0,0 +1,1012 @@
+                qb_names=[qbc.name, qbt.name],
+                options_dict={'TwoD': True, 'plot_all_traces': plot_all_traces,
+                              'plot_all_probs': plot_all_probs,
+                              'channel_map': channel_map})
+            cphases = flux_pulse_tdma.proc_data_dict[
+                'analysis_params_dict']['cphase']['val']
+            population_losses = flux_pulse_tdma.proc_data_dict[
+                'analysis_params_dict']['population_loss']['val']
+            leakage = flux_pulse_tdma.proc_data_dict[
+                'analysis_params_dict']['leakage']['val']
+            return cphases, population_losses, leakage, flux_pulse_tdma
+        else:
+            return
+
+    def measure_dynamic_phases(self, qbc, qbt, cz_pulse_name, hard_sweep_params=None,
+                               qubits_to_measure=None, cal_points=True,
+                               analyze=True, upload=True, n_cal_points_per_state=1,
+                               cal_states='auto', prep_params=None,
+                               exp_metadata=None, classified=False, update=False,
+                               reset_phases_before_measurement=True,
+                               prepend_n_cz=0):
+
+        if qubits_to_measure is None:
+            qubits_to_measure = [qbc, qbt]
+        if hard_sweep_params is None:
+            hard_sweep_params = {
+                'phase': {
+                    'values': np.tile(np.linspace(0, 2 * np.pi, 6) * 180 / np.pi, 2),
+                    'unit': 'deg'}}
+
+        if reset_phases_before_measurement:
+            dyn_phases = {qb.name: 0 for qb in qubits_to_measure}
+            self.get_pulse_par(cz_pulse_name,
+                               qbc, qbt, 'basis_rotation')(dyn_phases)
+        else:
+            dyn_phases = self.get_pulse_par(cz_pulse_name,
+                                            qbc, qbt, 'basis_rotation')()
+
+        qbc_name = qbc.name
+        qbt_name = qbt.name
+
+        # check whether qubits are connected
+        self.check_connection(qbc, qbt)
+
+        if prep_params is None:
+            prep_params = self.get_prep_params([qbc, qbt])
+
+        for qb in qubits_to_measure:
+            label = f'Dynamic_phase_measurement_CZ{qbt.name}{qbc.name}-{qb.name}'
+            qb.prepare(drive='timedomain')
+            MC = qbc.instr_mc.get_instr()
+
+            if cal_points:
+                cal_states = CalibrationPoints.guess_cal_states(cal_states)
+                cp = CalibrationPoints.single_qubit(
+                    qb.name, cal_states, n_per_state=n_cal_points_per_state)
+            else:
+                cp = None
+
+            seq, hard_sweep_points = \
+                fsqs.dynamic_phase_seq(
+                    qb_name=qb.name, hard_sweep_dict=hard_sweep_params,
+                    operation_dict=self.get_operation_dict(),
+                    cz_pulse_name=cz_pulse_name + f' {qbc.name} {qbt.name}',
```

Comment:
Does this mean that the cz_pulse_name argument of the function is not the actual pulse name, but only the prefix? This seems to be different from the convention used in measure_cphase. Should be unified.

Reply:
@chellings this was resolved by you, right?

Reply:
I have finally decided to ignore it for now because I found out that this inconsistency exists also in other modules. We might at some point clean up the naming scheme throughout all modules. To distinguish between the prefix and the full name, we could either use cz_pulse_prefix and cz_pulse_name or, e.g., cz_pulse_name and cz_operation (or cz_op_code).